backend/services/office_hours/event.py

```diff
@@ -0,0 +1,679 @@
+                f"Could not find Office Hours event with id={oh_event_delta.id}"
+            )
+
+        section_member_entity = self._check_user_section_membership(
+            subject.id, oh_event_entity.office_hours_section_id
+        )
+
+        if section_member_entity.member_role == RosterRole.STUDENT:
+            raise PermissionError(
+                "User is a Student. Do not have permission to update an event."
+            )
+
+        # Update Delta Fields
+        if oh_event_delta.event_date is not None:
+            oh_event_entity.date = oh_event_delta.event_date
+
+        if oh_event_delta.start_time is not None:
+            oh_event_entity.start_time = oh_event_delta.start_time
+
+        if oh_event_delta.end_time is not None:
+            oh_event_entity.end_time = oh_event_delta.end_time
+
+        if oh_event_delta.location_description is not None:
+            oh_event_entity.location_description = oh_event_delta.location_description
+
+        if oh_event_delta.description is not None:
+            oh_event_entity.description = oh_event_delta.description
+
+        if oh_event_delta.type is not None:
+            oh_event_entity.type = oh_event_delta.type
+
+        if oh_event_delta.room is not None:
+            oh_event_entity.room_id = oh_event_delta.room.id
+
+        if oh_event_delta.oh_section is not None:
+            raise Exception("Updating Office Hours Event OH Section is Not Allowed.")
+
+        self._session.commit()
+
+        # Return the model
+        return oh_event_entity.to_model()
+
+    def delete(self, subject: User, oh_event: OfficeHoursEvent) -> None:
+        """Delete the specified office hours event.
+
+        Args:
+            subject (User): The user initiating the delete operation.
+            oh_event (OfficeHoursEvent): The office hours event to be deleted.
+
+        Returns:
+            None
+
+        Raises:
+            PermissionError: If the user attempting to delete the event is a student.
+            ResourceNotFoundException: If the specified office hours event does not exist.
+
+        """
+        section_member_entity = self._check_user_section_membership(
+            subject.id, oh_event.oh_section.id
+        )
+
+        if section_member_entity.member_role == RosterRole.STUDENT:
+            raise PermissionError(
+                "User is a Student. Do not have permission to delete an office hours event."
+            )
+
+        oh_event_entity = self._session.get(OfficeHoursEventEntity, oh_event.id)
+
+        if len(oh_event_entity.tickets) != 0:
+            raise Exception(
+                f"Ticket data exists for office hours event with id={oh_event.id}. Unable to delete event due to existing ticket data."
+            )
+
+        self._session.delete(oh_event_entity)
+        self._session.commit()
+
+        return None
+
+    def get_event_by_id(self, subject: User, oh_event_id: int) -> OfficeHoursEvent:
```

Comment:
We injected the subject here so that the endpoint would still require an authenticated user to access it but not necessarily a member of the event oh section. 

Moreover, since some of our other service functions rely on first fetching an event from this function (such as get_event_queue_stats() under /API/event), we would rather have the specific functions handle the permissions specific to their functionality.

Reply:
Oh ok, sounds good @meghansun322 !